app/assets/javascripts/map.js

```diff
@@ -80,39 +100,101 @@ class Map {
     this.removeAllControl.addEventListener('click', this.handleRemoveAllPolygons);
+    document.getElementById('doneSelectingArea').addEventListener('click', this.handleDoneSelecting);
+    document.getElementsByName("serviceExpeditionCost").forEach((element) => { 
+      element.addEventListener('click', this.handleExpeditionInfoClick); 
+    });
+    document.getElementById("addBag").addEventListener('click', this.handleAddBagClick);
+    document.getElementById("removeBag").addEventListener('click', this.handleRemoveBagClick);
+    document.getElementById("doneAddOns").addEventListener('click', this.handleDoneAddOns);
+    document.getElementById("quote_email").addEventListener('keyup', () => this.enableSubmitButton);
+    document.getElementById("quote_phone_number").addEventListener('keyup', () => this.enableSubmitButton);
+    document.getElementById('contact-info-form').addEventListener('submit', this.handlePaymentSubmit);
+    document.getElementById('closeSuccess').addEventListener('click', this.reloadPage);
   }
 
-  showAddressModal(){
-    const submitAddressModal = $('#submitAddressModal')
-    submitAddressModal.modal('show');
-    submitAddressModal.keyup(function() {
-      $('input[type=submit]').removeAttr('disabled').removeClass('disabled');
-    });
+  enableFindButton(){
+    document.getElementById('addressSubmit').classList.remove('disabled');
   }
   
   handleSearchSubmit(event) {
     event.preventDefault();
-    $('#submitAddressModal').modal('hide')
+    this.showNextSection('collapseFindAddress', 'collapseMap');
+    document.getElementById('checkMarkAddress').classList.remove('hidden');
+    
     if (this.marker) { this.marker.setMap(null) }
     if (this.polygons.length > 0) {
       this.polygons.forEach(p => { p.setMap(null) });
       this.polygons = [];
-      this.onPolygonsChanged(this.polygons);
+      this.handlePolygonChanged(this.polygons);
+    }
+    let address = document.getElementById('addressInput').value;
+    this.geocodeAddress(address);
+  }
+
+  showNextSection(collapseNode, showNode) {
+    document.getElementById(collapseNode).classList.remove('show');
+    document.getElementById(showNode).classList.add('show');
+  }
+
+  handlePolygonChanged() {
+    this.totalAreaInSqFt = this.convertToSqFt(this.aggregateAreaInMts());
+    this.subTotal = this.calculateSubTotal();
+    this.updateAreaOnSummary();
+    this.updateGrandTotal();
+    if (this.totalAreaInSqFt <= 0) {
+      document.getElementById('checkMarkArea').classList.add('hidden');
+      document.getElementById('doneSelectingArea').classList.add('disabled');
+    } else {
+      document.getElementById('checkMarkArea').classList.remove('hidden');
+      document.getElementById('doneSelectingArea').classList.remove('disabled');
+
     }
-    this.geocodeAddress();
+    document.getElementById('areaSelectHint').classList.remove('hidden');
+
+  }
+
+  convertToSqFt(totalAreaInMts) {
+    return totalAreaInMts * constants.SQ_FT_CONVERT;
```

Comment:
This isn't something really "configurable". I'm not sure it needs to be in the `constants` object. nitpick